virttest/utils_test/libvirt.py

```diff
@@ -584,6 +584,72 @@ def setup_or_cleanup_gluster(is_setup, vol_name, brick_path="", pool_name="",
         return ""
 
 
+def setup_or_cleanup_iptables_rules(rules, cleanup=False, remote=False):
+    """
+    Setup or cleanup for iptable rules, it can be locally or remotely
+    :param rules: list of rules
+    :param cleanup: Boolean value, true to cleanup, false to setup
+    :param remote: Boolean value, true for remote, false for local clean or set
+    """
+    commands = []
+    # check the existing iptables rules in remote or local machine
+    iptable_check_cmd = "iptables -S"
+    if remote:
+        server_ip = params.get("server_ip")
+        server_user = params.get("server_user", "root")
+        server_pwd = params.get("server_pwd")
+        server_session = remote.wait_for_login('ssh', server_ip, '22',
+                                               server_user, server_pwd,
+                                               r"[\#\$]\s*$")
+        cmd_output = server_session.cmd_status_output(iptable_check_cmd)
+        if (cmd_output[0] == 0):
+            exist_rules = cmd_output[1].strip().split('\n')
+        else:
+            server_session.close()
+            raise error.TestError("iptables fails for command remotely %s",
+                                  iptable_check_cmd)
+    else:
+        try:
+            cmd_output = process.system_output(iptable_check_cmd, shell=True)
+            exist_rules = cmd_output.strip().split('\n')
+        except process.CmdError, info:
+            raise error.TestError("iptables fails for command locally %s",
+                                  iptable_check_cmd)
+    # check rules whether it is really needed to be added or cleaned
+    for rule in rules:
+        flag = False
+        if cleanup:
+            if (rule in (exist_rule for exist_rule in exist_rules)):
```

Comment:
iptables -S returns exist_rules list of [-P <rule1>, -N <rule2>, -A <rule3>], if I directly iterate with rule in exist_rules it will returns false even for existing rule.